lib/index.js

```diff
@@ -311,7 +310,7 @@ exports.decodePayload = function (data, binaryType, callback) {
       }
 
       var ret = callback(packet, i + n, l);
-      if (false === ret) return;
+      if (ret) return;
```

Comment:
The meaning has changed here, hasn't it? I think the `ret` response is used [here](https://github.com/socketio/engine.io-client/blob/1.8.2/lib/transports/polling.js#L131) to end the decoding earlier.

(it seems a test is missing :-) )